From ff74b53e04a5ab94cad947f4b4d63b3cedd81902 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Thu, 22 Apr 2021 09:32:37 -0400
Subject: [PATCH 10/23] Add constructor(s) for ut::Location

This patch resolves the following error with the MSVC compiler:

dict0dict.h(1096): error C2440: '<function-style-cast>': cannot convert from 'initializer list' to 'ut::Location'
dict0dict.h(1203): note: No constructor could take the source type, or constructor overload resolution was ambiguous
dict0dict.h(1203): note: This diagnostic occurred in the compiler generated function 'void dict_sys_t::for_each_table(Functor &)'
---
 storage/innobase/include/ut0core.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/storage/innobase/include/ut0core.h b/storage/innobase/include/ut0core.h
index 6d123183..8545f65b 100644
--- a/storage/innobase/include/ut0core.h
+++ b/storage/innobase/include/ut0core.h
@@ -33,6 +33,8 @@
 
 namespace ut {
 struct Location {
+  Location() {}
+  Location(const char *fn, size_t ln) : filename(fn), line(ln) {}
   const char *filename;
   size_t line;
   std::string str() const;
